src/EnergyPlus/AirflowNetwork/src/Solver.cpp

```diff
@@ -12128,6 +12191,726 @@ namespace AirflowNetwork {
+                                        "size. Return Branch is calculated using velocity at 5m/s. Simulation continues...",
+                                        AirflowNetworkSimu.ErrIndexDuct,
+                                        hydraulicDiameter,
+                                        hydraulicDiameter);
+                                }
+                            }
+                        } else if (SolFla == -2) {
+                            ShowFatalError(
+                                m_state,
+                                "Duct Autosizing for Return Branch calculation failed: iteration limits exceeded. Return Branch is calculated "
+                                "using velocity at 5m/s.");
+                        }
+                        if (SolFla < 0) {
+                            ReturnBranchD = sqrt(4.0 * flowrate / 5.0 / 3.1415926) * factor;
+                        } else {
+                            ReturnBranchD = hydraulicDiameter * factor;
+                        }
+                        ReturnBranchArea = ReturnBranchD * ReturnBranchD / 4.0 * 3.1415926;
+                        Velocity = flowrate / ReturnBranchArea;
+                    }
+                    if (AirflowNetworkSimu.ductSizeMethod == AirflowNetworkSimuProp::DuctSizeMethod::VelocityAndLoss) {
+                        if (Velocity > AirflowNetworkSimu.DuctSizeMaxV) {
+                            ReturnBranchD = sqrt(4.0 * flowrate / AirflowNetworkSimu.DuctSizeMaxV / 3.1415926);
+                            ReturnBranchArea = ReturnBranchD * ReturnBranchD / 4.0 * 3.1415926;
+                            ShowWarningError(
+                                m_state, "AirflowNetwork Duct Sizing: Duct Sizing Method = PressureLossWithMaximumVelocity for Return Branch size");
+                            ShowContinueError(
+                                m_state,
+                                format("The Maximum Airflow Velocity at {:.1R} is less than calculated velosity at {:.1R} using PressureLoss",
+                                       AirflowNetworkSimu.DuctSizeMaxV,
+                                       Velocity));
+                            ShowContinueError(m_state, "..The Maximum Airflow Velocity is used to calculate Return Branch Diameter");
+                        }
+                    }
+                }
+            }
+        }
+        // Assign autosize values in Duct element
+        for (AFNLinkNum = 1; AFNLinkNum <= AirflowNetworkNumOfLinks; AFNLinkNum++) {
+            CompNum = AirflowNetworkLinkageData(AFNLinkNum).CompNum;
+            TypeNum = AirflowNetworkCompData(CompNum).TypeNum;
+            if (AirflowNetworkLinkageData(AFNLinkNum).DuctLineType == iDuctLineType::SupplyTrunk) {
+                DisSysCompDuctData(TypeNum).hydraulicDiameter = SupplyTrunkD;
+                DisSysCompDuctData(TypeNum).A = SupplyTrunkArea;
```

Comment:
`auto &disSysCompDuct = DisSysCompDuctData(TypeNum);` followed by `disSysCompDuct.` will make this code more readable.